OpenRA.Mods.Common/Lint/CheckMapMetadata.cs

```diff
@@ -10,26 +10,30 @@
 #endregion
 
 using System;
+using OpenRA.Mods.Common.MapFormats;
 using OpenRA.Server;
 
 namespace OpenRA.Mods.Common.Lint
 {
 	public class CheckMapMetadata : ILintMapPass, ILintServerMapPass
 	{
-		void ILintMapPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Map map)
+		void ILintMapPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, IMap imap)
 		{
-			Run(emitError, map.MapFormat, map.Author, map.Title, map.Categories);
+			var map = (Map)imap;
+			var defaultMap = imap as DefaultMap;
+			Run(emitError, map.Author, map.Title, map.Categories, defaultMap.MapFormat);
 		}
 
-		void ILintServerMapPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, MapPreview map, Ruleset mapRules)
+		void ILintServerMapPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, MapPreview mapPreview, Ruleset mapRules)
 		{
-			Run(emitError, map.MapFormat, map.Author, map.Title, map.Categories);
+			Run(emitError, mapPreview.Author, mapPreview.Title, mapPreview.Categories, mapPreview.MapFormat);
 		}
 
-		void Run(Action<string> emitError, int mapFormat, string author, string title, string[] categories)
+		void Run(Action<string> emitError, string author, string title, string[] categories, int mapFormat)
 		{
-			if (mapFormat < Map.SupportedMapFormat)
-				emitError($"Map format {mapFormat} does not match the supported version {Map.CurrentMapFormat}.");
+			if (mapFormat != DefaultMap.SupportedMapFormat)
+				emitError("Map format {0} does not match the supported version {1}."
+					.F(mapFormat, DefaultMap.SupportedMapFormat));
```

Comment:
Revert these changes

Reply:
Have brought it back however I am not sure what `CurrentMapFormat` means since there is simply `MapFormat` in the new version which is checked against the `SupportedMapFormat`. May need some clarification on this.